app/models/service_template_ansible_tower.rb

```diff
@@ -24,7 +24,13 @@ def self.create_catalog_item(options, _auth_user = nil)
 
   def remove_invalid_resource
     # remove the resource from both memory and table
-    service_resources.to_a.delete_if { |r| r.destroy unless r.reload.resource.present? }
+    service_resources.to_a.delete_if do |r|
+      if r.id
```

Comment:
I think this would read better if we'd check `if r.persisted?`.  Or `if r.new_record?` and reverse the logic.

Reply:
@jrafanie will you please approve, now?

Reply:
this is a 25+ comment PR with less than 20 functional lines of code, you know?